iac-extensions/arm/src/test/resources/checks/EmptyOrNullValueCheckTest/emptyOrNullValue.bicep

```diff
@@ -0,0 +1,112 @@
+//^^^^^^^^
+  tags: {}                // Noncompliant {{Remove this empty object or complete with real code.}}
+//^^^^^^^^
+  dependsOn: []           // Noncompliant {{Remove this empty array or complete with real code.}}
+//^^^^^^^^^^^^^
+  properties: {
+    prop1: null           // Noncompliant {{Remove this null property or complete with real code.}}
+    prop2: ''             // Noncompliant {{Remove this empty string or complete with real code.}}
+    prop3: {}             // Noncompliant {{Remove this empty object or complete with real code.}}
+    prop4: []             // Noncompliant {{Remove this empty array or complete with real code.}}
+    prop5: 'my string'
+    prop6: identifier
+    prop7: true
+    prop8: {key:'val'}
+    prop9: ['val']
+    prop10: [
+      null,      // Don't report an issue on empty/null value, only if it's the value of a property
+      '',
+      [],
+      {},
+      {
+        key: null // Noncompliant {{Remove this null property or complete with real code.}}
+      }
+    ]
+  }
+  other: [
+    {
+      key:null    // Noncompliant {{Remove this null property or complete with real code.}}
+    }
+  ]
+}
+
+var var1 = null           // Noncompliant {{Remove this null variable or complete with real code.}}
+var var2 = ''             // Noncompliant {{Remove this empty string or complete with real code.}}
+var var3 = {}             // Noncompliant {{Remove this empty object or complete with real code.}}
+var var4 = []             // Noncompliant {{Remove this empty array or complete with real code.}}
+
+var var5 = 'my string'
+var var6 = identifier
+var var7 = true
+var var8 = {key:'val'}
+var var9 = ['val']
+var var10 = [
+  {
+    key: null               // Noncompliant {{Remove this null property or complete with real code.}}
+  }
+]
+
+output out1 object = null // Noncompliant {{Remove this null output or complete with real code.}}
+output out2 string = ''   // Noncompliant {{Remove this empty string or complete with real code.}}
+output out3 object = {}   // Noncompliant {{Remove this empty object or complete with real code.}}
+output out4 array = []    // Noncompliant {{Remove this empty array or complete with real code.}}
+
+output out5 string = 'my string'
+output out6 string = identifier
+output out7 bool = true
+output out8 object = {key:'val'}
+output out9 array = ['val']
+output out10 object = {
+  key:null                // Noncompliant {{Remove this null property or complete with real code.}}
+}
+output out11 array = [
+  null,
+  '',
+  [],
+  {},
+  {
+    key:null              // Noncompliant {{Remove this null property or complete with real code.}}
+  }
+]
+
+output outIf object = 1 > 2 ? null : '' // Not reporting any issue in case of conditional output
+
+param par1 object = null
+param par2 string = ''
+param par3 object = {}
+param par4 array = []
+param myList array = [
+  'val1'
+  'val2'
+  'val3'
+]
```

Comment:
1. Those params are between `output`s, please move before or after
2. Maybe we should also raise issues for empty/null parameters? 🤔  The value is optional, the default value needs to be verified for different types. WDYT?

Reply:
1. 👍 
2. That's a good question., initially I was thinking that it it was not needed for param, but I don't see a valid use case with null/empty value. I will give a try and see if there is a lot of issues raised in ITs.

Reply:
I checked. This is not a good idea as there can be a valid use case: an optional parameter with an empty value by default.
If no default value is provided, then providing a value on deployment is mandatory; otherwise, it will fail.